rust/lancedb/src/connection.rs

```diff
@@ -873,9 +881,19 @@ impl Database {
 
                 let registry = Arc::new(ObjectStoreRegistry::default());
                 let storage_options = options.storage_options.clone();
-                let os_params = ObjectStoreParams {
-                    storage_options: Some(storage_options.clone()),
-                    ..Default::default()
+                let os_params = if let Some(os) = &options.object_store {
+                    ObjectStoreParams {
+                        storage_options: Some(storage_options.clone()),
+                        object_store: Some((os.0.clone(), url)),
+                        use_constant_size_upload_parts: os.1.is_some(),
+                        block_size: os.1,
```

Comment:
Sorry for the confusion. `block_size` is for reads. It determines when range requests are coalesced.

`use_constant_size_upload_parts` is for writes. It determines whether multipart uploads only use fixed-size parts, or variable size parts (higher max object size).

These two shouldn't be controlled by the same parameter.